### PR TITLE
[MI-169] Use properties for resource name and add a default object name

### PR DESCRIPTION
**Pull request by miogalang**
Proposed pattern for setting object names.

/cc @jwswj @joseconsador @mmolina @atroche

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-169

### Risks
* None

### CODE REVIEW

File: src/Zendesk/API/Resources/TicketFields.php [hunk 1]
```diff
@@ -10,8 +10,5 @@ class TicketFields extends ResourceAbstract
 {
     use Defaults;
 
-    const OBJ_NAME = 'ticket_field';
```
Comment (miogalang) [inline src/Zendesk/API/Resources/TicketFields.php]: Example for default pattern with two words

File: src/Zendesk/API/Resources/Automations.php [hunk 2]
```diff
@@ -11,9 +11,6 @@
  */
 class Automations extends ResourceAbstract
 {
-    const OBJ_NAME = 'automation';
```
Comment (miogalang) [inline src/Zendesk/API/Resources/Automations.php]: Pattern for default

File: src/Zendesk/API/Resources/TicketImports.php [hunk 3]
```diff
@@ -18,6 +18,15 @@ class TicketImports extends ResourceAbstract
     use CreateMany;
 
     /**
+     * @var String
+     */
+    protected $objectName = 'ticket';
```
Comment (miogalang) [inline src/Zendesk/API/Resources/TicketImports.php]: Pattern for custom pattern where we override it in the resource class

File: src/Zendesk/API/Resources/ResourceAbstract.php [hunk 4]
```diff
@@ -50,6 +59,14 @@ public function __construct(HttpClient $client)
             $this->resourceName = $this->getResourceNameFromClass();
         }
 
+        if (! isset($this->objectName)) {
+            $this->objectName = $this->getResourceNameFromClass();
+        }
+
+        if (! isset($this->objectNamePlural)) {
+            $this->objectNamePlural = $this->getResourceNameFromClass() . 's';
```
Comment (jwswj) [inline src/Zendesk/API/Resources/ResourceAbstract.php]: ewww

**Comment by miogalang** [comment]
Updated to use inflector for pluralizing and singularizing.

**Comment by jwswj** [comment]
Nice - I'm a fan. @joseconsador ?

**Comment by joseconsador** [comment]
:+1:

**Comment by miogalang** [comment]
ready for review :haircut: @joseconsador @jwswj

File: src/Zendesk/API/Resources/Core/Brands.php [hunk 5]
```diff
@@ -18,9 +18,7 @@
  */
 class Brands extends ResourceAbstract
 {
-    const OBJ_NAME = 'brand';
-    const OBJ_NAME_PLURAL = 'brands';
-
+    use Defaults;
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Core/Brands.php]: duplicate?

File: src/Zendesk/API/Resources/Core/JobStatuses.php [hunk 6]
```diff
@@ -11,10 +11,16 @@
  */
 class JobStatuses extends ResourceAbstract
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Core/JobStatuses.php]: Should this class be JobStatus?

File: src/Zendesk/API/Resources/Core/SlaPolicies.php [hunk 7]
```diff
@@ -11,10 +11,13 @@
  */
 class SlaPolicies extends ResourceAbstract
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Core/SlaPolicies.php]: Should this be SlaPolicy ?

File: src/Zendesk/API/Resources/Core/JobStatuses.php (hunk 6 above)
Comment (miogalang) [inline src/Zendesk/API/Resources/Core/JobStatuses.php]: We use the plural form of the word for the resource names. It was the standard Joms and I agreed on

**Comment by jwswj** [comment]
:+1:

**Pull request merged**